test/parallel/test-buffer-slice.js

```diff
@@ -62,3 +62,15 @@ assert.strictEqual(Buffer.alloc(0).slice(0, 1).length, 0);
 
 // slice(0,0).length === 0
 assert.strictEqual(0, Buffer.from('hello').slice(0, 0).length);
+
+{
+  // Test included in https://github.com/nodejs/node/issues/9096
+  // If the offset or ending value provided is a floating point number, it
+  // should be coerced to integer consistently. Otherwise the actual length of
+  // the slice calculation will be messed up.
```

Comment:
This comment seems a little confusing to me. The offsets *are* consistently converted to integers (in C++). The real problem is that the `end` value is calculated in javascript land *before* it gets passed to C++. If the `length` calculation (used to determine `end`) was done inside C++ this wouldn't be an issue.

It's also possible to use floating point numbers with `slice()` and still get a reasonable result. The only issue is when the `end` argument is not supplied.